src/Type/MockReturnType.php

```diff
@@ -74,27 +75,48 @@ private function getTypeFromCall(
 
         $arg = $args[0]->value;
 
-        if (!$arg instanceof ClassConstFetch) {
+        $classes = [];
+
+        if ($arg instanceof Array_) {
+            foreach ($arg->items as $item) {
+                if ($item->value instanceof ClassConstFetch) {
+                    $classes[] = $this->getTypeFromClassConst($item->value, $scope);
+                }
+            }
+        }
+
+        if ($arg instanceof ClassConstFetch) {
+            $classes[] = $this->getTypeFromClassConst($arg, $scope);
+        }
+
+        $classes = array_unique(array_filter($classes));
+
+        if (!$classes) {
             return $reflection->getReturnType();
         }
 
-        $class = $arg->class;
+        return new InstanceHandleType(...$classes);
+    }
+
+    private function getTypeFromClassConst(ClassConstFetch $classConstFetch, Scope $scope): ?string
```

Comment:
This doesn't really return a "type" in terms of PHPStan terminology. A better name might be `getNameFromClassConst` or `getClassNameFromConst`.

Also, this repo currently supports PHP 7.0, so nullable types can't be used unfortunately (this is why the Travis build is failing btw).